Werkzeug für die aktuell eingestellten Mailvorlagen

schnumsel opened the pull request:
Hallo Matthias,

also, vielleicht mal zum drüberschaun. Habe das Modul so erweitert, dass unter System -> Tools ein E-Mail-Preview Tool existiert, mit dem man die aktuellen Mails anschaun kann, also auch solche, die z.B. mit Yireo-E-Mail-Override aus Templates und nicht aus der DB erzeugt werden.

Viele Grüße,

Willi

schnumsel committed:
Update composer.json

To be able to deploy it

wilfriedwolf committed:
Prepare tool for Magento email development

wilfriedwolf committed:
Reverting to original composer.json

wilfriedwolf committed:
Coninue

wilfriedwolf committed:
Merge branch 'master' of ssh://gitlab.sandstein.de/bodenbelag/marketplace

# Conflicts:
#	app/code/local/Bodenbelag/Marketplace/Model/Dropship/Vendor/Statement.php
#	app/code/local/Bodenbelag/Marketplace/Model/DropshipPaypalAdaptive/Adaptive.php
#	app/code/local/Bodenbelag/Marketplace/Model/Observer.php
#	app/code/local/Bodenbelag/Marketplace/etc/config.xml

Moving additional Templates from bodenbelag/transaction-mails to bodenbelag/theme (using magento-snm/transaction-mails instead)

wilfriedwolf committed:
Merge branch 'master' of ssh://gitlab.sandstein.de/bodenbelag/marketplace

# Conflicts:
#	app/code/local/Bodenbelag/Marketplace/Model/Dropship/Vendor/Statement.php
#	app/code/local/Bodenbelag/Marketplace/Model/DropshipPaypalAdaptive/Adaptive.php
#	app/code/local/Bodenbelag/Marketplace/Model/Observer.php
#	app/code/local/Bodenbelag/Marketplace/etc/config.xml

Moving additional Templates from bodenbelag/transaction-mails to bodenbelag/theme (using magento-snm/transaction-mails instead)

wilfriedwolf committed:
Merge branch 'master' of ssh://gitlab.sandstein.de/bodenbelag/marketplace

# Conflicts:
#	app/code/local/Bodenbelag/Marketplace/Model/Dropship/Vendor/Statement.php
#	app/code/local/Bodenbelag/Marketplace/Model/DropshipPaypalAdaptive/Adaptive.php
#	app/code/local/Bodenbelag/Marketplace/Model/Observer.php
#	app/code/local/Bodenbelag/Marketplace/etc/config.xml

Moving additional Templates from bodenbelag/transaction-mails to bodenbelag/theme (using magento-snm/transaction-mails instead)

sprankhub commented on composer.json:
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "magento-hackathon/e-mail-preview",
+    "name": "schnumsel/e-mail-preview",
```
I think this should still be `magento-hackathon`.

schnumsel replied:
Of course, your right. I did this, not knowing, i could define another branch to fetch the schnumsel stuff via composer.

sprankhub commented on .gitignore:
```diff
@@ -0,0 +1 @@
+# Created by .ignore support plugin (hsz.mobi)
```
This file should be removed.

schnumsel replied:
Well, I like to have .gitignore in order not to accidentially commit .idea folder. How do you guys handle this? Via IDE-Settings?

schmengler replied:
My PhpStorm project root is never the Git repository itself.

Also helpful: http://stackoverflow.com/questions/5724455/can-i-make-a-user-specific-gitignore-file

edannenberg replied:
Personally I would consider a user specific `.gitignore` bad practise, who cares about a couple of extra lines in a `.gitignore` file. Not that there aren't use cases for it, but generally speaking.

@schnumsel The current `.gitignore` is effectively empty though, might as well not have it.

sprankhub replied:
The `.idea` folder is in the project root, the extension is usually in a subfolder. Hence, no need for such a `.gitignore`. Additionally, as said by @edannenberg, it is empty anyway and can/should be removed.

mzeis commented:
Hi Willi,

thank you for your contribution!

If you still would like this to be merged please can you improve the things mentioned by Simon and push again to your branch so I can have a look?

bambamboole commented:
@schnumsel It would be really nice if you could improve the things because the feature is super useful!

schnumsel committed:
Changes for Pull Request Werkzeug für die aktuell eingestellten Mailvorlagen #19

schnumsel commented:
Well, there it is. Sorry I did the changes in the master-branch not in the dev-willi-branch. Well finally it is done.
Best regards,
Willi

mzeis commented on app/code/community/Hackathon/EmailPreview/etc/config.xml:
```diff
@@ -181,6 +387,22 @@
+                        <method>hackathonEmailpreviewRenderEmailBefore</method>
+                    </hackathon_emailpreview_contact>
+                    <hackathon_emailpreview_admin_password_forgot>
+                        <class>hackathon_emailpreview/mail_type_adminPasswordForgot</class>
```
This class is missing in the PR, causing a warning (and therefore error in developer mode). Please can you add it?

schnumsel replied:
I'll try to check it up these days, maybe next friday since I have a day off...

mzeis commented on app/code/community/Hackathon/EmailPreview/Model/Mail/Type/AdminPasswordEmail.php:
```diff
@@ -0,0 +1,27 @@
+            return $this;
+        }
+
+        //@todo change logic here
```
What would be necessary to get this todo done (question also applies for the other @todo comments)?

schnumsel replied:
same here. quite a long time, since i changed the code  :)

mzeis reviewed (changes requested):
Thank you very much for updating!

I reviewed your PR and found one problem causing the new functionality to break.

Please let's also discuss whether we should implement the open todos marked in the code or remove the comments. As I don't use this extension in any project anymore I might not finish these todos myself. :-)

schnumsel committed:
Finishing missing class and todos for pull request "Werkzeug für die aktuell eingestellten Mailvorlagen #19"

schnumsel committed:
Corrected wrong class reference in config.xml (not the class was missing, only the reference was wrong)
Implemented Oauth and System Errors Mails (got rid of the todos)
Removed product alert in favor of the existing product alert price and product alert stock observer classes.

schnumsel commented:
So hello again,

I promised Friday and I managed it (well 12 Minutes left :)
Hope that's it. Haven't delt a long time with this module..

Best regards,

Willi

mzeis commented:
Thank you, that's commitment! :-) I will be away for two weeks but I'll do the review afterwards.

mzeis commented:
Thank you again  very much for your contribution! The changes are now in the `develop` branch.

Had to do some dirty changes to get it working with the dev branch quickly but will clean up a bit before releasing the new version.

schnumsel commented:
Well, unfortunately the dirty hacks did actually break something. My "Hackathon_EmailPreview_Block_Adminhtml_Email_Preview" is quite different compared to the final one.
I am missing specially the Increment ID Field, no order testing without.
How could we proceed?
Best regards, Willi

mzeis commented:
I'm sorry about the Increment ID field - I fixed this now in dcb8ac8. The other PR introduced dependencies between fields, hiding the form element.

> My "Hackathon_EmailPreview_Block_Adminhtml_Email_Preview" is quite different compared to the final one.

Actually that's not because of the "hack" (the one I meant I already removed), that's because the `master` branch you branched from was quite different from the `dev` branch where other PRs had been merged before.

Of course, the goal should be to provide a stable v1.1.0. Even if the code is not perfect existing functionality shouldn't break though of course, that's important.

For a future release, it might make sense to do more refactoring, maybe drop the dependency on PHP 5.3 to allow to profit from new languages feature. As I honestly can't spare much time for this I cannot give a guarantee when I might do that.
